my_project/tab_psy_chart/app_psy_chart.py

```diff
@@ -308,26 +313,25 @@ def update_psych_chart(
     elif var == "Frequency":
         var_color = ["rgba(255,255,255,0)", "rgb(0,150,255)", "rgb(0,0,150)"]
     else:
-        var_unit = mapping_dictionary[var]["unit"]
+        var_unit = mapping_dictionary[var][si_ip]["unit"]
 
         var_name = mapping_dictionary[var]["name"]
 
         var_color = mapping_dictionary[var]["color"]
 
     if global_local == "global":
         # Set Global values for Max and minimum
-        var_range_x = mapping_dictionary["DBT"]["range"]
-        hr_range = [0, 0.03]
-        var_range_y = hr_range
+        var_range_x = mapping_dictionary["DBT"][si_ip]["range"]
+        var_range_y = mapping_dictionary["hr"][si_ip]["range"]
 
     else:
         # Set maximum and minimum according to data
         data_max = 5 * ceil(df["DBT"].max() / 5)
         data_min = 5 * floor(df["DBT"].min() / 5)
         var_range_x = [data_min, data_max]
 
-        data_max = (5 * ceil(df["hr"].max() * 1000 / 5)) / 1000
-        data_min = (5 * floor(df["hr"].min() * 1000 / 5)) / 1000
+        data_max = round(df["hr"].max(),4)
+        data_min = round(df["hr"].min(),4)
```

Comment:
I changed the max and min data acquisition methods. The previous method may lead to some deviation when the unit system changes.